packages/scm/src/browser/scm-widget.tsx

```diff
@@ -149,6 +149,7 @@ export class ScmWidget extends BaseWidget implements StatefulWidget {
 
     protected onActivateRequest(msg: Message): void {
         super.onActivateRequest(msg);
+        this.refresh();
```

Comment:
I think the real issue is that `ScmRepository.onDidChange` does not fire then `GitWatcher.onGitEvent` reports a change event for the repository.